sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/NumberConverter.scala

```diff
@@ -89,6 +91,10 @@ object NumberConverter {
     var i = fromPos
     while (i < value.length) {
       value(i) = Character.digit(value(i), radix).asInstanceOf[Byte]
+      // if invalid characters are found, it no need to convert the suffix starting there
+      if (value(i) < 0) {
```

Comment:
do we need to do `value(i) = -1` here? Otherwise the doc is wrong: `..., convert it to -1 and ...`

Reply:
Character.digit(value(i), radix).asInstanceOf[Byte] will return -1 if the character is invalid

Reply:
then shall we add `assert(value(i) == -1)`?

Reply:
if you like, but I don't think it needed since Character.digit(value(i), radix) is a simple function and it guaranteed to return -1 if character is invalid in its docs

![image](https://user-images.githubusercontent.com/84778052/127167938-d97e934e-dc42-4139-87d6-051b839d368c.png)